### PR TITLE
[refactor-wip](TaskWorkerPool)add specific classes for tasks

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes

Issue Number: #19575 

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run p0

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by bobhan1** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.66 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/%Y%m%d%H%M%S_clickbench_pr_7.html

**Comment by hello-stephen** [comment]
(From new mechine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 54.83 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          88.5 seconds inserted 10000000 Rows, about 112K ops/s
 storage size: 17167658531 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230705104212_clickbench_pr_172390.html

**Comment by bobhan1** [comment]
run p0

**Comment by hello-stephen** [comment]
(From new mechine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 55.32 seconds
 stream load tsv:          504 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          85.7 seconds inserted 10000000 Rows, about 116K ops/s
 storage size: 17167184935 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230705132553_clickbench_pr_172532.html

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by bobhan1** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.44 seconds
 stream load tsv:          516 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162091972 Bytes

File: be/src/agent/task_worker_pool.cpp [hunk 1]
```diff
@@ -1963,4 +1745,239 @@ Status StorageMediumMigrateTaskPool::_check_migrate_request(const TStorageMedium
     return Status::OK();
 }
 
+CheckConsistencyTaskPool::CheckConsistencyTaskPool(ExecEnv* env, ThreadModel thread_model)
```
Comment (gavinchou) [inline be/src/agent/task_worker_pool.cpp:1748]: It's better to reduce duplicated code that declares `XxxTaskPool` multiple times by invoking the template.
Here is an example, the implementation of each callback (template specialization) can be put into a separate file
img_v2_668c21f7-9817-43d2-ace1-1516f44843cg

**Pull request closed without merging**